src/main/java/org/interledger/InterledgerAddress.java

```diff
@@ -109,12 +133,151 @@ default boolean startsWith(final InterledgerAddress interledgerAddress) {
    * <p> Returns the hash code value for this <tt>InterledgerAddress</tt>. The
    * <tt>InterledgerAddress</tt> interface is essentially a type-safe wrapper around a String value,
    * so implementations should take care to forward hashcode decisions to the
-   * {@link String#equals(Object)} method on the object returned from {@link #getValue()}.</p>
+   * {@link String#equals(Object)} method on the object returned of {@link #getValue()}.</p>
    *
    * @return the hash code value for this InterledgerAddress.
+   *
    * @see Object#hashCode()
    * @see Object#equals(Object)
    */
   @Override
   int hashCode();
+
+  /**
+   * A builder for immutable instances of {@link InterledgerAddress}.
+   *
+   * <p> <em>NOTE: {@code Builder} is not thread-safe and generally should not be stored in a field
+   * or collection, but instead used immediately to create instances.</em> </p>
+   */
+  class Builder {
+
+    private String value;
+
+    /**
+     * No-args Constructor.
+     */
+    private Builder() {
+    }
+
+    /**
+     * Builder method to actually construct an instance of {@link InterledgerAddress} of the data
+     * in this builder.
+     */
+    public InterledgerAddress build() {
+      return new Impl(this);
+    }
+
+    /**
+     * Assign a new value to this builder.
+     *
+     * @param value A {@link String} representing this builder's "value", which is the string
+     *              version of an Interledger Address.
+     */
+    public Builder value(final String value) {
+      this.value = value;
```

Comment:
I've moved most of the checks into the `build()` function per discussion with @sappenin.

The idea is that the builder should only throw when you call build so you can provide it with objects that may change between creating the builder and calling build.